### PR TITLE
Move along pca fixes

### PR DESCRIPTION
**Pull request by sheryjoe**

### CODE REVIEW

File: Prep/Source/MeshTools/ParticleBasedSurfaceReconstruction/source/ReconstructSamplesAlongPCAModes.cxx [hunk 1]
```diff
@@ -77,31 +77,44 @@ int DoIt(InputParams params)
...
+                {
+                    NumberOfModes = n;
+                    found = true;
+                }
+            }
 
-    if(!found)
-        NumberOfModes = percentVarByMode.size();
+            if(!found)
+                NumberOfModes = percentVarByMode.size();
```
Comment (prafulag) [inline Prep/Source/MeshTools/ParticleBasedSurfaceReconstruction/source/ReconstructSamplesAlongPCAModes.cxx]: why is this needed after line 101 ?

Comment (prafulag) [reply Prep/Source/MeshTools/ParticleBasedSurfaceReconstruction/source/ReconstructSamplesAlongPCAModes.cxx]: is it just a fail safe for 101 ?

File: Prep/Source/MeshTools/ParticleBasedSurfaceReconstruction/source/ReconstructSamplesAlongPCAModes.cxx [hunk 2]
```diff
@@ -130,17 +143,28 @@ int DoIt(InputParams params)
     {
```
Comment (prafulag) [inline Prep/Source/MeshTools/ParticleBasedSurfaceReconstruction/source/ReconstructSamplesAlongPCAModes.cxx]: This loop may fail when we provide the particular mode index, unless the mode to be generated is very first mode, since NumberOfModes is set to 1.

File: Prep/Source/MeshTools/ParticleBasedSurfaceReconstruction/source/ReconstructSamplesAlongPCAModes.cxx (hunk 1 above)
Comment (sheryjoe) [reply Prep/Source/MeshTools/ParticleBasedSurfaceReconstruction/source/ReconstructSamplesAlongPCAModes.cxx]: yes

File: Prep/Source/MeshTools/ParticleBasedSurfaceReconstruction/source/ReconstructSamplesAlongPCAModes.cxx (hunk 2 above)
Comment (sheryjoe) [reply Prep/Source/MeshTools/ParticleBasedSurfaceReconstruction/source/ReconstructSamplesAlongPCAModes.cxx]: fixed

Comment (prafulag) [reply Prep/Source/MeshTools/ParticleBasedSurfaceReconstruction/source/ReconstructSamplesAlongPCAModes.cxx]: I don't think it is fixed yet, coz the NumberOfModes is set to 1, so loop runs only one time. The continue condition was definitely needed, but the NumberOfModes need to be set to max_possible_number_of_modes. Since we already have the flag for singleModeToBeGen, number of modes need not be set to 1.
